### PR TITLE
Dm_Char01

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 1]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
+        this->unk_34C = 2;
+        this->actionFunc = func_80AA8F1C;
+    }
+}
+
+void func_80AA8F0C(DmChar01* this, GlobalContext* globalCtx) {
+}
+
+void func_80AA8F1C(DmChar01* this, GlobalContext* globalCtx) {
+}
```
Comment (isghj5) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:326]: So weird, it has two do nothing functions...

**Review by isghj5 — APPROVED** [review]
Nothing major stopping me from approving

File: assets/xml/overlays/ovl_Dm_Char01.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<Root>
+    <File Name="ovl_Dm_Char01" BaseAddress="0x80aa81e0" RangeStart="0x1BE0" RangeEnd="0x28B0">
```
Comment (engineer124) [inline assets/xml/overlays/ovl_Dm_Char01.xml:2]:
```suggestion
    <File Name="ovl_Dm_Char01" BaseAddress="0x80AA81E0" RangeStart="0x1BE0" RangeEnd="0x28B0">
```

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 3]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AAAAB0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 300, ICHAIN_STOP),
 };
 
-#endif
+static s16 D_80AAAAB4 = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:53]:
```suggestion
static s16 D_80AAAAB4 = false;
```
This variable is boolean, you can change throughout

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 4]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
+            this->unk_348 = 255.0f;
+
+            for (i = 0; i < ARRAY_COUNT(this->unk_1AC); i++) {
+                this->unk_1AC[i] = ovl_dm_char01_Vtx_1BE0[i].v.ob[1] * 409.6f;
+            }
+
+            DynaPolyActor_Init(&this->dyna, 0);
+            DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_mtoride_Colheader_009E4C);
+
+            this->unk_34D = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:88]:
```suggestion
            this->unk_34D = true;
```
This is also boolean and can be changed

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 5]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char01/func_80AA9020.s")
+            if (D_80AAAE24 != 0) {
+                if ((D_80AAAE22 > -1800) && (D_80AAAE22 < 3000)) {
+                    temp_f12 = D_80AAAE22 - 640.0f;
+                    if ((D_80AAAE20 == 380) && (D_80AAAE22 > 640)) {
+                        D_80AAAAC0 = 2;
+                        D_80AAAAC4 = 0;
+                        D_80AAAAC8 = 900;
+                        D_80AAAACC = 700;
+                        spB7 = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:497]:
```suggestion
                        spB7 = true;
```
Also boolean

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 6]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
+                        if (D_80AAAE22 < 1350) {
+                            f32 temp_f0 = temp_f12 / 2000.0f;
+                            D_80AAAAB8 = 420.0f - (420.0f * temp_f0);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:500]:
```suggestion
                            f32 temp_f0 = temp_f12 / 2000.0f;
                            
                            D_80AAAAB8 = 420.0f - (420.0f * temp_f0);
```

File: src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c [hunk 7]
```diff
@@ -28,44 +46,498 @@ const ActorInit Dm_Char01_InitVars = {
...
+                            D_80AAAABC = (200.0f * temp_f0) + 200.0f;
+                        } else {
+                            f32 temp_f0 = temp_f12 / 2000.0f;
+                            D_80AAAAB8 = 420.0f - (420.0f * temp_f0);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Dm_Char01/z_dm_char01.c:504]:
```suggestion
                            f32 temp_f0 = temp_f12 / 2000.0f;
                            
                            D_80AAAAB8 = 420.0f - (420.0f * temp_f0);
```

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**